acousticbrainz/data/dataset.py

```diff
@@ -0,0 +1,214 @@
+import psycopg2
+from flask import current_app
+from werkzeug.exceptions import BadRequest, ServiceUnavailable
+
+
+DATASET_JSON_SCHEMA = {
+    "type": "object",
+    "properties": {
+        "name": {
+            "type": "string",
+            "minLength": 1,
+            "maxLength": 100
+        },
+        "description": {"type": "string"},
+        "classes": {
+            "type": "array",
+            "minItems": 2,  # must have at least two classes
+            "items": {
+                # CLASS
+                "type": "object",
+                "properties": {
+                    "name": {
+                        "type": "string",
+                        "minLength": 1,
+                        "maxLength": 100
+                    },
+                    "description": {"type": "string"},
+                    "recordings": {
+                        # CLASS_MEMBER
+                        "type": "array",
+                        "minItems": 1,  # must have at one recording
+                        "items": {"type": "string"},  # FIXME: This should be a UUID
+                    },
+                },
+            },
+        },
+    },
+}
+
+
+def create_from_dict(dictionary, author_id=None):
+    """Creates a new dataset from a dictionary.
+
+    Data in the dictionary must be validated using `DATASET_JSON_SCHEMA` before
+    being passed to this function.
+
+    Returns:
+        Tuple with two values: new dataset ID and error. If error occurs first
+        will be None and second is an exception. If there are no errors, second
+        value will be None.
+    """
+    try:
+        connection = psycopg2.connect(current_app.config["PG_CONNECT"])
+        cursor = connection.cursor()
+
+        cursor.execute("""INSERT INTO dataset (id, name, description, author)
```

Comment:
I'd like to see this split into a few different methods for reuse, and testability (you have tests, right? ;)